hyperspy/_signals/spectrum.py

```diff
@@ -84,7 +84,10 @@ def _spikes_diagnosis(self, signal_mask=None,
             dc = dc[~navigation_mask, :]
         der = np.abs(np.diff(dc, 1, -1))
         plt.figure()
-        plt.hist(np.ravel(der.max(-1)), 100)
+        if len(dc.shape) == 1:
```

Comment:
Ok, new histograms: https://imgur.com/a/AfpK6

You can see that since there are just a few spikes, the overall scale of the plot is dominated by the "no spikes" region that is so tall, making it hard to see the spiked points. They are slightly visible in the current implementation (first image), and in a lot of the `get_histogram` methods, they are not visible at all due to the binning.